framework/base/Event.php

```diff
@@ -187,9 +251,20 @@ class_implements($class, true)
      */
     public static function trigger($class, $name, $event = null)
     {
-        if (empty(self::$_events[$name])) {
+        $wildcardEventHandlers = [];
+        if (!empty(self::$_eventWildcards)) {
+            foreach (self::$_eventWildcards as $nameWildcard => $classHandlers) {
+                if (!StringHelper::matchWildcard($nameWildcard, $name)) {
+                    continue;
+                }
+                $wildcardEventHandlers = array_merge($wildcardEventHandlers, $classHandlers);
+            }
+        }
+
+        if (empty(self::$_events[$name]) && empty($wildcardEventHandlers)) {
```

Comment:
Conditions could be inverted. Checking single variable costs less then accessing array index.

Reply:
optimized